src/EditorFeatures/Core/Implementation/ForegroundNotification/ForegroundNotificationService.cs

```diff
@@ -139,6 +147,15 @@ private void NotifyOnForegroundWorker()
                         {
                             // eat up cancellation
                         }
+                        catch (Exception ex) when (FatalError.ReportWithoutCrash(ex))
+                        {
+                            // The PendingWork callbacks should never throw.  In the case they do we
+                            // must ensure the IAsyncToken implementation is completed.  If it is not
+                            // then the unit test code will end up in a deadlock doing an 'await' 
+                            // on the token instance.
+                            Debug.Assert(false, ex.Message);
+                            done = true;
```

Comment:
@CyrusNajmabadi agree it's unlikely for it to be `false` here.  Pedantically `Enqueue` or `UpdateToLocalTime` could throw in which case we'd end up here with a `done == false`.  Given the trouble it took to find this deadlock I err'd on the side of over cautious here + future proofing.  Can remove if desired.